README.md

```
@@ -19,6 +19,46 @@ Start developing by following these steps:
+#### Compare Metrics Tests
+
+The `TestCompareMetrics` test is a test that compares the generated metrics from a running controller with the expected metrics defined in the `testdata`.
+
+```bash
+make test_compare_metrics
+```
+
+Per default the tests will use [`envtest`](https://github.com/kubernetes-sigs/controller-runtime/tree/main/pkg/envtest) to start a temporary Kubernetes API server and `etcd` instance. If you want to run the tests against an existing cluster, you can use the `envtest` supported environment variable `USE_EXISTING_CLUSTER` and set it to `true`.
+
+> [!WARNING] In this case, the current context of your `kubeconfig` will be used to connect to the cluster.
+
```

markdownlint-cli2 (0.22.1)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)